src/routes/blog/+page.server.ts

```diff
@@ -0,0 +1,64 @@
+import * as fs from 'fs'
+import * as readline from 'node:readline'
+
+export interface PageServerLoadBlog {
+  articles: [{
+    filename: string
+    birthtime: date
+  }]
+}
+
+interface Metadata {
+  title: string
+  creationDate: Date
+}
+
+async function parseMetaData (pathToFile): Promise<Metadata> {
+  return await new Promise<Metadata>((resolve, reject) => {
+    let lineCounter = 0
+    const metadata: Metadata = {}
+    const streamToFile = fs.createReadStream(pathToFile)
+    const lineReader = readline.createInterface({ input: streamToFile })
+
+    lineReader.on('line', (line) => {
+      if (lineCounter > 1) {
+        lineReader.close()
+      } else {
+        metadata[line.split(':')[0]] = line.split(':')[1]
+      }
+      lineCounter++
+    })
+
+    lineReader.on('close', function () {
+      streamToFile.close()
+      resolve(metadata)
+    })
+  })
+}
+
+async function getArticles (): Promise<object[]> {
```

Comment:
suggestion: If you type correctly `articles`, you can correct type here too

```suggestion
async function getArticles (): Promise<PageServerLoadBlog['articles'][]> {
```